### ISSUE TITLE
Run Sample Algorithm Using the run_algorithm() Interface

### ISSUE THREAD

**Issue by fredfortier** [enigmampc/catalyst#26]
I'm able to successfully run the buy_and_hodl.py (note typo) sample algorithm using the command line interface as suggested in the documentation.

However, I also need to ability to run it using the run_algorithm() interface for more control during the execution.

Here is my adapted version of the algorithm:
```python
from catalyst.utils.run_algo import run_algorithm
from datetime import datetime
import pytz

from catalyst.api import (
    order_target_value,
    symbol,
    record,
    cancel_order,
    get_open_orders,
)


def initialize(context):
    context.ASSET_NAME = 'USDT_BTC'
    context.TARGET_HODL_RATIO = 0.8
    context.RESERVE_RATIO = 1.0 - context.TARGET_HODL_RATIO

    # For all trading pairs in the poloniex bundle, the default denomination
    # currently supported by Catalyst is 1/1000th of a full coin. Use this
    # constant to scale the price of up to that of a full coin if desired.
    context.TICK_SIZE = 1000.0

    context.is_buying = True
    context.asset = symbol(context.ASSET_NAME)

    context.i = 0


def handle_data(context, data):
    context.i += 1

    print 'i:', context.i

    starting_cash = context.portfolio.starting_cash
    target_hodl_value = context.TARGET_HODL_RATIO * starting_cash
    reserve_value = context.RESERVE_RATIO * starting_cash

    # Cancel any outstanding orders
    orders = get_open_orders(context.asset) or []
    for order in orders:
        cancel_order(order)

    # Stop buying after passing the reserve threshold
    cash = context.portfolio.cash
    if cash <= reserve_value:
        context.is_buying = False

    # Retrieve current asset price from pricing data
    price = data[context.asset].price

    # Check if still buying and could (approximately) afford another purchase
    if context.is_buying and cash > price:
        # Place order to make position in asset equal to target_hodl_value
        order_target_value(
            context.asset,
            target_hodl_value,
            limit_price=price * 1.1,
            stop_price=price * 0.9,
        )

    record(
        price=price,
        cash=cash,
        starting_cash=context.portfolio.starting_cash,
        leverage=context.account.leverage,
    )


start = datetime(2015, 3, 1, 0, 0, 0, 0, pytz.utc)
end = datetime(2017, 6, 28, 0, 0, 0, 0, pytz.utc)
run_algorithm(
    initialize=initialize,
    handle_data=handle_data,
    start=start,
    end=end,
    capital_base=100000,
    bundle='poloniex'
)
```

Here is the output:
```
[2017-08-16 17:11:57.438224] INFO: Loader: Loading benchmark data for 'USDT_BTC' from 2015-02-28 00:00:00+00:00 to 2017-08-14 00:00:00+00:00
[2017-08-16 17:11:57.478054] WARNING: Loader: Refusing to download new treasury data because a download succeeded at 2017-08-16 16:19:53+00:00.
Traceback (most recent call last):
  File "catalyst/examples/buy_and_hold_live.py", line 78, in <module>
    bundle='poloniex'
  File "/Users/fredfortier/Code/enigma-projects/catalyst-baseline/catalyst/utils/run_algo.py", line 414, in run_algorithm
    environ=environ,
  File "/Users/fredfortier/Code/enigma-projects/catalyst-baseline/catalyst/utils/run_algo.py", line 197, in _run
    env, data = get_trading_env_and_data(bundles)
  File "/Users/fredfortier/Code/enigma-projects/catalyst-baseline/catalyst/utils/run_algo.py", line 168, in get_trading_env_and_data
    adjustment_reader=bundle_data.adjustment_reader,
  File "/Users/fredfortier/Code/enigma-projects/catalyst-baseline/catalyst/data/data_portal.py", line 319, in __init__
    if self._first_trading_day is not None else (None, None)
  File "/Users/fredfortier/Code/enigma-projects/catalyst-baseline/catalyst/utils/calendars/trading_calendar.py", line 695, in open_and_close_for_session
    sched.at[session_label, 'market_open'].tz_localize('UTC'),
  File "/Users/fredfortier/Code/enigma-projects/catalyst-baseline/python2/lib/python2.7/site-packages/pandas/core/indexing.py", line 1665, in __getitem__
    return self.obj.get_value(*key, takeable=self._takeable)
  File "/Users/fredfortier/Code/enigma-projects/catalyst-baseline/python2/lib/python2.7/site-packages/pandas/core/frame.py", line 1900, in get_value
    return engine.get_value(series.get_values(), index)
  File "pandas/index.pyx", line 98, in pandas.index.IndexEngine.get_value (pandas/index.c:3557)
  File "pandas/index.pyx", line 106, in pandas.index.IndexEngine.get_value (pandas/index.c:3240)
  File "pandas/index.pyx", line 583, in pandas.index.DatetimeEngine.get_loc (pandas/index.c:11566)
KeyError: Timestamp('2012-02-01 00:00:00+0000', tz='UTC')
```

Do you get the same error on your side?

**Comment by fredfortier** [comment]
I figured it out. I was able to run this algo in my end.

**Comment by fredfortier** [comment]
I closed this one too fast.

The problem seems to be a discrepancy between the "enigma-catalyst" project on pip and the sources in the master branch.

What's confusing is that the error does not occur when running the algorithm from the command-line:
```
python catalyst/examples/buy_and_hold_live.py
```
Intuitively, I expected python to run using the project sources. However, I could empirically validate that it is instead loading at least "run_algo.py" from elsewhere (presumably the "enigma-catalyst" client). I came to this conclusion by deliberately introducing errors and new log statements in "run_algo.py".

PyCharm, on the other hand, correctly loads "run_algo.py" from sources. I confirmed this by observing my additional log statements in the output. This condition results in the error in the original post.

In summary, I believe that there is a real error condition in the master branch. In addition, there is likely a discrepancy in the "enigma-catalyst" project on pip. I'm not sure if there is a way to further isolate the "enigma-catalyst" project to ensure that python always prioritize the project sources.

**Comment by lacabra** [comment]
@fredfortier the error that you report in the original post is due to a version mismatch between the data bundle and the code, which has now been fixed in the latest release (0.1.dev8).

Yes, the master branch and pip (which uses the Python Package Index, PyPI for short) were out of sync for several weeks. That's what I had been working on in reorganizing our several branches. They were brought back in sync on Aug 18, 2017. Your second post dates from a couple of days earlier arguing they were out of sync. Yes, they were then, but they are back in sync now (and that's why it's important that no commits are made to master, except for when we make a new release (that then gets pushed to PyPI).

I have tried your code above, and runs equally well on catalyst and directly using run_algorithm(). Thus, I'm closing this issue. Feel free to open it if you still have problems after upgrading to 0.1.dev8.
